variables.tf

```diff
@@ -0,0 +1,8 @@
+variable "deployment_id" {
+  description = "Creates a random string that will be used in tagging for correlating the resources used with a deployment of AAP."
+  type    = string
+  validation {
+    condition     = length(var.deployment_id) == 8
```

Comment:
Should there be more basic conditions? This would allow `!@#$%^&*` which I would guess isn't valid in AWS for an id.